### ISSUE TITLE
B3 headers are not considered when there are case-miss matches

### ISSUE THREAD

**Issue by iampratapak** [helidon-io/helidon#4956]
Are B3 header names case-sensitive? B3 propagation is not happening when cases mismatch. Incoming B3 headers are not considered for tracing when cases miss matches.

## Environment Details
* Helidon Version: 3.0.1
* Helidon SE or Helidon MP: Helidon SE
* JDK version: Java 17
* OS: Windows 11
* Docker version (if applicable):

----------

## Problem Description
Incoming B3 headers are not considered for tracing when cases miss matches. If B3 headers exactly match those below, then it's working; otherwise, it's not.
Working:
`X-B3-TraceId, X-B3-SpanId, X-B3-ParentSpanId, X-B3-Sampled`
Not Working:
`x-b3-traceid, x-b3-spanid, x-b3-parentspanid, x-b3-sampled`
`X-B3-TraceID, X-B3-SpANId, X-B3-PARENTSPANID, x-b3-SAMPLED`
and so on...

## Steps to reproduce
Create a Helidon SE 3.0.1 application and configure Jaeger tracing with B3 propagation.
```
tracing:
  service: helidon-quick-start-1
  protocol: http
  port: 14250
  path: /api/traces
  tags:
    env: development
  enabled: true
  sampler-type: "const"
  sampler-param: 1
  log-spans: true
  propagation: b3
```
Add B3 headers to the request, all in small case.
<img width="487" alt="image" src="https://user-images.githubusercontent.com/12915303/192179238-d5c1a638-d4eb-4270-bd8a-08acc8e1c0f0.png">
Incoming B3 header trace ID should be used, and the same should be propagated to downstream applications, but a new trace ID is generated and propagated to the downstream application.
<img width="401" alt="image" src="https://user-images.githubusercontent.com/12915303/192179573-1b2d3968-4b06-4a94-b04e-a18021cce0a2.png">
<img width="473" alt="image" src="https://user-images.githubusercontent.com/12915303/192179628-ed4829ce-5ece-4566-9033-c3d8398100d9.png">

Headers need to be case insensitive.

**Comment by pratapgowda007** [comment]
@tjquinno 
When case-mismatches occur in the case of w3c headers, incorrectly propagated header values are also detected. 

traceparent --> working as expected
traceParent --> not working as expected

As you can see from the below images, propagated span-id is different in uber-trace-id and transparent.

<img width="167" alt="image" src="https://user-images.githubusercontent.com/12915303/196419494-3b64e6bc-9b26-4e60-a655-99e02a8621a8.png">

<img width="642" alt="image" src="https://user-images.githubusercontent.com/12915303/196418935-4396c042-a4e1-4f6d-9942-343c2d5951b6.png">

![image](https://user-images.githubusercontent.com/12915303/196419117-77e7ef00-8951-4ae3-acdb-53b4a5c09479.png)

<img width="274" alt="image" src="https://user-images.githubusercontent.com/12915303/196419199-d3bfbe2e-813b-4ab6-a1dd-d2ab8c7145dd.png">

**Comment by tjquinno** [comment]
Yes, there are general issues with case sensitivity and headers, not just in tracing. Please see #5149 which I am actively working on.

It looks as if fixing that issue will take care of the B3 and W3C tracing issues, but I have added specific tracing tests to be sure.

**Comment by pratapgowda007** [comment]
@tjquinno thanks for the update.

**Comment by tjquinno** [comment]
Fixed by #5221
